dev/src/bulk-writer.ts

```diff
@@ -668,14 +815,21 @@ export class BulkWriter {
    *
    * @private
    */
-  private getEligibleBatch<T>(): BulkCommitBatch {
-    if (this.batchQueue.length > 0) {
-      const lastBatch = this.batchQueue[this.batchQueue.length - 1];
-      if (lastBatch.state === BatchState.OPEN) {
+  private getEligibleBatch<T>(retryCount: number): BulkCommitBatch {
```

Comment:
If we don't need the count here, we can pass in a boolean (or even the queue we would use)